changelog.d/20251006_162634_kevin_clear_environ_immediately_after_fork.rst

```diff
@@ -0,0 +1,6 @@
+Bug Fixes
+^^^^^^^^^
+
+- Clear environment immediately after ``fork()``, prior even to dropping
+  privileges.  (The environment was already cleared prior to ``exec()`` ing the
+  child UEP, but there's no sense in waiting during the pre-exec phase.)
```

Comment:
As discussed out-of-band, we should update this to address the bug in which the UEP inherits the MEP's `GLOBUS_COMPUTE_USER_DIR` environment variable and uses it to create its own endpoint directory, which often fails due to filesystem permissions.